uamqp/management_link.py

```diff
@@ -126,27 +128,39 @@ def _on_message_received(self, message):
                 to_remove_operation = operation
                 break
         if to_remove_operation:
-            mgmt_result = ManagementExecuteOperationResult.OK \
-                if 200 <= status_code <= 299 else ManagementExecuteOperationResult.FAILED_BAD_STATUS
             to_remove_operation.on_execute_operation_complete(
-                mgmt_result,
+                ManagementExecuteOperationResult.OK,
                 status_code,
                 status_description,
                 message,
                 response_detail.get(b'error-condition')
             )
             self._pending_operations.remove(to_remove_operation)
 
-    def _on_send_complete(self, message, reason, state):
+    def _on_send_complete(self, message, reason, state):  # todo: reason is never used, should check spec
         if SEND_DISPOSITION_REJECT in state:  # either rejected or accepted
             to_remove_operation = None
             for operation in self._pending_operations:
                 if message == operation.message:
                     to_remove_operation = operation
                     break
             self._pending_operations.remove(to_remove_operation)
+            # TODO: better error handling
+            error_response = ErrorResponse(error_info=state[SEND_DISPOSITION_REJECT])
+            # The callback is defined in management_operation.py
+            # TODO: AMQPException is too general? to be more specific: MessageReject(Error) or AMQPManagementError?
+            #  or should there an error mapping which maps the condition to the error type
```

Comment:
With regards to how to handle the error - starting general is fine. 
You will probably find while adapting the EH and SB library that you need finer grain control of errors - then we can usually add more specific subclasses that should be non-breaking.

Reply:
sounds good!